scipy/stats/_hypotests.py

```diff
@@ -1682,145 +1682,6 @@ def cramervonmises_2samp(x, y, method='auto'):
-        .. [1] NIST/SEMATECH e-Handbook of Statistical Methods, "7.4.7.1.
-               Tukey's Method."
-               https://www.itl.nist.gov/div898/handbook/prc/section4/prc471.htm,
-               28 November 2020.
-
-        Examples
-        --------
-        >>> from scipy.stats import tukey_hsd
-        >>> group0 = [24.5, 23.5, 26.4, 27.1, 29.9]
-        >>> group1 = [28.4, 34.2, 29.5, 32.2, 30.1]
-        >>> group2 = [26.1, 28.3, 24.3, 26.2, 27.8]
-        >>> result = tukey_hsd(group0, group1, group2)
-        >>> ci = result.confidence_interval()
-        >>> ci.low
-        array([[-3.649159, -8.249159, -3.909159],
-               [ 0.950841, -3.649159,  0.690841],
-               [-3.389159, -7.989159, -3.649159]])
-        >>> ci.high
-        array([[ 3.649159, -0.950841,  3.389159],
-               [ 8.249159,  3.649159,  7.989159],
-               [ 3.909159, -0.690841,  3.649159]])
-        """
-        # check to see if the supplied confidence level matches that of the
-        # previously computed CI.
-        if (self._ci is not None and self._ci_cl is not None and
-                confidence_level == self._ci_cl):
-            return self._ci
-
-        if not 0 < confidence_level < 1:
-            raise ValueError("Confidence level must be between 0 and 1.")
-        # determine the critical value of the studentized range using the
-        # appropriate confidence level, number of treatments, and degrees
-        # of freedom as determined by the number of data less the number of
-        # treatments. ("Confidence limits for Tukey's method")[1]. Note that
-        # in the cases of unequal sample sizes there will be a criterion for
-        # each group comparison.
-        params = (confidence_level, self._nobs, self._ntreatments - self._nobs)
-        srd = distributions.studentized_range.ppf(*params)
-        # also called maximum critical value, the Tukey criterion is the
-        # studentized range critical value * the square root of mean square
-        # error over the sample size.
-        tukey_criterion = srd * self._stand_err
-        # the confidence levels are determined by the
-        # `mean_differences` +- `tukey_criterion`
-        upper_conf = self.statistic + tukey_criterion
-        lower_conf = self.statistic - tukey_criterion
-        self._ci = ConfidenceInterval(low=lower_conf, high=upper_conf)
-        self._ci_cl = confidence_level
-        return self._ci
-
-
-def _tukey_hsd_iv(args):
-    if (len(args)) < 2:
-        raise ValueError("There must be more than 1 treatment.")
-    args = [np.asarray(arg) for arg in args]
-    for arg in args:
-        if arg.ndim != 1:
-            raise ValueError("Input samples must be one-dimensional.")
-        if arg.size <= 1:
-            raise ValueError("Input sample size must be greater than one.")
-        if np.isinf(arg).any():
-            raise ValueError("Input samples must be finite.")
-    return args
-
-
 def tukey_hsd(*args):
```

Comment:
This should be moved, too, right?
Before we submit the SciPy PR, let's re-do this so that the commit history of these lines is preserved. I described the process of creating copies of files and merging them to preserve history when we were discussing the reorganization of stats. I think we can take that for a test drive here.

Reply:
So TBH I think while we're reviewing this offline it would be easier to just create a file `_multicomp` without the movement.